### PR TITLE
Replace Facebook Graph API with Google Calendar File for Events

### LINKED ISSUE #116: Google Calendar Integration
Marketing is in the process of migrating away from Facebook events to Instagram and Discord, so to make it easier, we should set up a Google Calendar of events and query this with the Google Calendar API instead.

Specifically this will query an ICS file of all our events and that will be parsed and displayed on the page.

Note: This should be completed after #115

### LINKED ISSUE #117: Facebook Events API no longer returns aggregated events
Discovered by a potential general member on September 25:

The Events page was unable to load events because the API proxy did not receive any events. We initially suspected the API key had expired, but after investigation, we discovered the API key was still active but providing only the club page ID instead of any events. CloudWatch logs indicate this has occurred since September 12 which coincides with [the release of Facebook Graph API v18.0](https://developers.facebook.com/blog/post/2023/09/12/introducing-facebook-graph-and-marketing-api-v18/), although there is no indication that the events API would be deprecated. The API is still able to query data of individual events given an event ID but cannot provide all events.

We considered temporarily using a static payload of events, but the Facebook image URLs are only temporarily signed and will eventually expire (should be after one day). We could find another image hosting solution, but the events payload would need to be manually updated whenever marketing posts a new event.

Given the delays in the site migration to Next.js and Sanity (currently on [HackAtUCI/hack.ics.uci.edu](https://github.com/HackAtUCI/hack.ics.uci.edu)), we may need to temporarily pursue the Google Calendar integration suggested in #116.

Related Stack Overflow question: [Facebook events graph API stopped working(Sep 2023)?](https://stackoverflow.com/questions/77086231/facebook-events-graph-api-stopped-workingsep-2023)

### PR DESCRIPTION
**Pull request by taesungh**
Replace Facebook Graph API with Google Calendar
- Since Facebook's API stopped working for aggregate events data, fetch events data from a public Google Calendar file using node-ical
- Acquire file attachment link for images from Google Drive
- Update event data properties on client and remove Facebook link button
- Remove references to `FB_PAGE_TOKEN`

Configure local API testing with proxy to express
- Create dev server with express to locally test API endpoints
- Use http-proxy-middleware since native proxy option is broken
- Make deployment action skip installing dev dependencies
- Audit client dependencies

Fixes #117 and closes #116 ahead of #115.

### CODE REVIEW

**Comment by taesungh** [comment]
Need to fix event ordering and filtering.

File: client/src/app/containers/event-card/EventCard.js [hunk 1]
```diff
@@ -89,20 +88,10 @@ function EventCard(props) {
...
-              className='popup-button-event-card-link'
-            >
-              <span>View on Facebook</span>
-            </a>
-          </button>
         </div>
         <div className='desc-text-wrapper-event-card'>
           <p className='desc-text-event-card'>
-            <Linkify> {props.description}</Linkify>
+            <div dangerouslySetInnerHTML={{ __html: props.description }} />
```
Comment (samderanova) [inline client/src/app/containers/event-card/EventCard.js:94]: Question: was there a reason for using `dangerouslySetInnerHTML` instead of Linkify?

**Review by samderanova — COMMENTED** [review]
Thanks for all your hard work towards fixing these issues! One small question and then, as you stated, once you get ordering and filtering to work, we should be good to go.

File: client/src/app/containers/event-card/EventCard.js (hunk 1 above)
Comment (taesungh) [reply client/src/app/containers/event-card/EventCard.js:82]: Google Calendar provides the event description as HTML tags from their available formatting options (bold, italic, underline, unordered list, ordered list, and links). react-linkify was only for turning link text into anchors and cannot parse other HTML elements.

File: client/src/app/containers/event-card/EventCard.js [hunk 2]
```diff
@@ -89,20 +76,10 @@ function EventCard(props) {
...
           </p>
         </div>
         <div className='modal-bottom-part'>
```
Comment (samderanova) [inline client/src/app/containers/event-card/EventCard.js:85]: Nitpick: would prefer to have a transition on modal open and close as well as button hover, but if it's too much, no worries.

Comment (samderanova) [reply client/src/app/containers/event-card/EventCard.js:85]: Issue: The modal font isn't Poppins. Was there a reason for keeping it as the Bootstrap default?

Comment (samderanova) [reply client/src/app/containers/event-card/EventCard.js:85]: Issue: the modal scrollbar is somewhat hard to see due to low contrast. Is it possible to configure it?
image

**Review by samderanova — CHANGES_REQUESTED** [review]
A few additional suggestions regarding the modal:

File: client/src/app/containers/event-card/EventCard.js (hunk 2 above)
Comment (taesungh) [reply client/src/app/containers/event-card/EventCard.js:85]: While I agree these are important lingering design issues, they are outside the scope of the data ingestion changes happening with this PR. Let's create a separate issue for improving the EventCard modal design.

**Review by samderanova — APPROVED** [review]
Sounds good. Let's merge and then, as you said, we can create new issue(s) to address them.

**Pull request merged**